modules.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * skogul, module automation utilities
+ *
+ * Copyright (c) 2019 Telenor Norge AS
+ * Author(s):
+ *  - Kristian Lyngstøl <kly@kly.no>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package skogul
+
+// Module is metadata for a skogul module. It is used by the receiver,
+// sender and transformer package. The Alloc() function must return a
+// data structure that implements the relevant module interface, which is
+// checked primarily in config.Parse.
+//
+// See */auto.go for how to utilize this, and config/help.go,
+// cmd/skogul/main.go for how to extract information/help, and ultimately
+// config/parse.go for how it is applied.
+type Module struct {
+	Name    string             // short name of the module (e.g: "http"
+	Aliases []string           // optional aliases (e.g. "https")
+	Alloc   func() interface{} // allocation of a blank module structure
```

Comment:
This is the only thing I'm a bit reluctant about.. Losing out on a bit of type safety/"clarity". But I guess that's a price to pay when missing generics 😅

Reply:
Yeah, I know. We could conceivably have type safety with reflection, but I don't think it's worth it. We still get implicit type safety because the moment you try to load a module it is cast and checked, so it's impossible to use (or test) the wrong type.